docs/troubleshooting.md

```diff
@@ -202,3 +202,120 @@ If you're using v1.10.0, `aws-node` daemonset pod requires IMDSv1 access to obta
 
 See the [cni-metrics-helper README](../cmd/cni-metrics-helper/README.md).
 
+
+## Debugging Unit Test Failures on PR submission
+
+### Run unit tests / Unit-tests (pull_request)
+The above workflow gets triggered for every PR. It setups up a self hosted github runner with ubuntu:latest image and runs the unit tests. 
+To debug failures, click on details and check the phase in which the workflow failed. Expand the failed phase and check for the failed command. Try one of the ways to debug further
+
+- Run the failed command on your machine  
+OR  
+- Provision an EC2 instance with ubuntu:latest image (as used by self hosted runner)  
```

Comment:
Provision an EC2 instance with ubuntu:latest image (as it is used by github hosted runner used for build and unit test execution.)